modules/javafx.graphics/src/main/java/com/sun/javafx/text/TextRun.java

```diff
@@ -393,7 +393,7 @@ public int getGlyphAtX(float x, int[] trailing) {
         float runX = 0;
         for (int i = 0; i < glyphCount; i++) {
             float advance = getAdvance(i);
-            if (runX + advance > x) {
+            if (runX + advance >= x) {
```

Comment:
this is very interesting.  how did it work before?
could there be any scenarios that might have failed before and are fixed with this change?

Reply:
Without this condition when `runX + advance` becomes equal to `x`, character index calculated will be wrong. 
For example if we have 2 emojis (😀😃), when we move from first emoji to second, character index returned will be 1 instead of 2 when we just hit the second emoji. This is a failing scenario.